commons-statistics-distribution/src/main/java/org/apache/commons/statistics/distribution/TruncatedNormalDistribution.java

```diff
@@ -0,0 +1,206 @@
+        double cdfBeta = standardNormal.cumulativeProbability(beta);
+        cdfDelta = cdfBeta - cdfAlpha;
+
+        // Calculation of variance and mean.
+        double pdfAlpha = standardNormal.density(alpha);
+        double pdfBeta = standardNormal.density(beta);
+        double pdfCdfDelta = (pdfAlpha - pdfBeta) / cdfDelta;
+        double alphaBetaDelta = (alpha * pdfAlpha - beta * pdfBeta) / cdfDelta;
+
+        if (lower == Double.NEGATIVE_INFINITY && upper != Double.POSITIVE_INFINITY) {
+            // One-sided lower tail truncation
+            double betaRatio = pdfBeta / cdfBeta;
+            this.mean = mean - sd * betaRatio;
+            variance = sd * sd * (1 - beta * betaRatio - betaRatio * betaRatio);
+        } else if (lower != Double.NEGATIVE_INFINITY && upper == Double.POSITIVE_INFINITY) {
+            // One-sided upper tail truncation
+            double alphaRatio =  pdfAlpha / cdfDelta;
+            this.mean = mean + sd * alphaRatio;
+            variance = sd * sd * (1 + alpha * alphaRatio - alphaRatio * alphaRatio);
+        } else if (lower != Double.NEGATIVE_INFINITY && upper != Double.POSITIVE_INFINITY) {
+            // Two-sided truncation
+            this.mean = mean + pdfCdfDelta * parentSd;
+            variance = sd * sd * (1 + alphaBetaDelta - pdfCdfDelta * pdfCdfDelta);
+        } else {
+            // No truncation
+            this.mean = mean;
+            variance = sd * sd;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public double density(double x) {
+        if (x < lower || x > upper) {
+            return 0;
+        }
+        return standardNormal.density((x - parentMean) / parentSd) / (parentSd * cdfDelta);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public double cumulativeProbability(double x) {
+        if (x <= lower) {
+            return 0;
+        } else if (x >= upper) {
+            return 1;
+        } else {
+            return (standardNormal.cumulativeProbability((x - parentMean) / parentSd) - cdfAlpha) / cdfDelta;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public double inverseCumulativeProbability(double p) {
+        if (p < 0 || p > 1) {
+            throw new DistributionException(DistributionException.INVALID_PROBABILITY, p);
+        }
+        return standardNormal.inverseCumulativeProbability(cdfAlpha + p * cdfDelta) * parentSd + parentMean;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Represents the true mean of the truncated normal distribution rather
+     * than the parent normal distribution mean.
+     */
+    @Override
+    public double getMean() {
+        return mean;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Represents the true variance of the truncated normal distribution rather
+     * than the parent normal distribution variance.
+     */
+    @Override
+    public double getVariance() {
+        return variance;
+    }
+
+    /**
```

Comment:
In this case where nothing is added to the javadoc the rest of the codebase uses a single line `/** {@inheritDoc} */`